File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.js
/**
 * @param {string} s
 * @param {number} k
 * @return {number}
 */
var maxVowels = function(s, k) {
    let i = 0
    let numVowels = 0
    let res = 0
    for (let j=0; j < s.length; j++) {
        if ("aeiou".includes(s[j])) numVowels++
        if (j - i + 1 === k) {
            res = Math.max(res, numVowels)
            if ('aeiou'.includes(s[i])) numVowels--
            i++
        }
    }
    return res
};

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        res = vowels = i = 0
        for j in range(0, len(s)):
            if s[j] in "aeiou":
                vowels += 1
            if j - i + 1 == k:
                res = max(res, vowels)
                if s[i] in "aeiou":
                    vowels -= 1
                i += 1
        return res